cmd/endpoint-ellipses.go

```diff
@@ -129,6 +112,44 @@ func getSetIndexes(args []string, totalSizes []uint64, customSetDriveCount uint6
 			return nil, config.ErrInvalidErasureSetSize(nil).Msg(msg)
 		}
 		setSize = customSetDriveCount
+		globalCustomErasureDriveCount = true
+	} else {
+		var newSetCounts = make(map[uint64]struct{})
+		for _, ss := range setCounts {
+			var valid bool
+			for _, argPattern := range argPatterns {
+				for _, p := range argPattern {
+					if uint64(len(p.Seq)) > ss {
+						valid = uint64(len(p.Seq))%ss == 0
+					} else {
+						valid = ss%uint64(len(p.Seq)) == 0
+					}
+				}
+			}
+			if _, ok := newSetCounts[ss]; !ok && (valid || argPatterns == nil) {
+				newSetCounts[ss] = struct{}{}
+			}
+		}
+
+		setCounts = []uint64{}
+		for setCount := range newSetCounts {
+			setCounts = append(setCounts, setCount)
+		}
+
+		if commonSize > setCounts[len(setCounts)-1] {
+			prevD := commonSize / setCounts[0]
+			for _, i := range setCounts {
```

Comment:
Let's not call it `i` so it doesn't get confused with an index. Maybe:

```suggestion
			for _, cnt := range setCounts {
```